### PR TITLE
Tsc 555  upload profile picture

### PR DESCRIPTION
**Pull request by paologum**
# Main

* made downloading a file more modular and easy to integrate temporary files
* can upload datapack profile pictures
* will store in the datapacks directory
* accepts only png, jpg, and jpeg
* front end and back end support for the file types
* added new route to service these images

https://github.com/user-attachments/assets/56855413-c3b0-41f7-b330-b4bac3235e55

https://github.com/user-attachments/assets/26556231-031b-42ea-b11c-6c8be9b42bad

### CODE REVIEW

**Review by asivath — COMMENTED** [review]
can you edit the datapack profile images?

**Comment by paologum** [comment]
> can you edit the datapack profile images?

no editing, just uploading profile images, the title of the pr is misleading so i'll change it

File: server/src/upload-handlers.ts [hunk 1]
```diff
@@ -139,7 +142,8 @@ export async function setupNewDatapackDirectoryInUUIDDirectory(
   uuid: string,
   sourceFilePath: string,
   metadata: DatapackMetadata,
-  manual?: boolean // if true, the source file will not be deleted and admin config will not be updated in memory or in the file system
+  manual: boolean, // if true, the source file will not be deleted and admin config will not be updated in memory or in the file system
```
[the remark this answers is not in the document]
Comment (paologum) [reply server/src/upload-handlers.ts:145]: yup once everything is consistent in the new structure i'll do that in a different pr

File: server/src/routes/routes.ts [hunk 2]
```diff
@@ -437,3 +438,34 @@ export const fetchTimescale = async function (_request: FastifyRequest, reply: F
...
+      if (!(await checkFileExists(defaultFilepath))) {
+        reply.send({ error: "Default image not found" });
+        return;
+      }
+      reply.send(await readFile(defaultFilepath));
+      return;
+    }
+    const uniqueImageFilepath = await fetchDatapackProfilePictureFilepath(decodeURIComponent(uuid), title);
+    if (!(await checkFileExists(uniqueImageFilepath))) {
+      if (!(await checkFileExists(defaultFilepath))) {
```
[the remark this answers is not in the document]
Comment (paologum) [reply server/src/routes/routes.ts:459]: i'm confused where do we call it twice in a row without returning, if it is on line 450, it can't get to line 458

**Review by asivath — APPROVED** [review]

**Pull request merged**